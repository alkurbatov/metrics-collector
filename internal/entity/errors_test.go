package entity_test

import (
	"net"
	"net/http"
	"testing"

	"github.com/alkurbatov/metrics-collector/internal/entity"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestMetricNotImplementedError(t *testing.T) {
	err := entity.MetricNotImplementedError("summary")

	require.ErrorIs(t, err, entity.ErrMetricNotImplemented)
	snaps.MatchSnapshot(t, err.Error())
}

func TestHTTPError(t *testing.T) {
	err := entity.HTTPError(http.StatusInternalServerError, []byte("Something bad has happened"))

	require.ErrorIs(t, err, entity.ErrHTTP)
	snaps.MatchSnapshot(t, err.Error())
}

func TestEncodingNotSupportedError(t *testing.T) {
	err := entity.EncodingNotSupportedError("deflate")

	require.ErrorIs(t, err, entity.ErrEncodingNotSupported)
	snaps.MatchSnapshot(t, err.Error())
}

func TestTestUntrustedSourceError(t *testing.T) {
	tt := []struct {
		name     string
		sourceIP string
	}{
		{
			name: "Empty IP",
		},
		{
			name:     "IPv4 value",
			sourceIP: "192.168.10.12",
		},
		{
			name:     "IPv6 value",
			sourceIP: "2001:470:28:30d:21e:67ff:fe7a:e1da",
		},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			err := entity.UntrustedSourceError(net.ParseIP(tc.sourceIP))

			require.ErrorIs(t, err, entity.ErrUntrustedSource)
			snaps.MatchSnapshot(t, err.Error())
		})
	}
}
