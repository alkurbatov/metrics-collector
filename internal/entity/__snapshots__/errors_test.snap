
[TestMetricNotImplementedError - 1]
metric kind not supported (summary)
---

[TestHTTPError - 1]
HTTP request failed (500): Something bad has happened
---

[TestEncodingNotSupportedError - 1]
encoding type not supported (deflate)
---

[TestTestUntrustedSourceError/Empty_IP - 1]
request's source is not trusted (<nil>)
---

[TestTestUntrustedSourceError/IPv4_value - 1]
request's source is not trusted (192.168.10.12)
---

[TestTestUntrustedSourceError/IPv6_value - 1]
request's source is not trusted (2001:470:28:30d:21e:67ff:fe7a:e1da)
---

[TestTransportNotSupportedError - 1]
transport type not supported (tcp)
---
